### PR TITLE
Feature/4355 channel value rnpv5

### PR DESCRIPTION
**Pull request by surajeaton**
Fixes # BLUI-4355

#### Changes proposed in this Pull Request:
- Fixed Channel Value icon issue
- Updated channel value text variants
- Added test case

#### Screenshots / Screen Recording (if applicable)
- 
Simulator Screenshot - iPhone 14 - 2023-09-13 at 16 12 21

#### To Test:
- cd MD3
- yarn ios

#### Any specific feedback you are looking for?
-

### CODE REVIEW

File: components/src/md3/ChannelValue/ChannelValue.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React, { useCallback } from 'react';
 import { View, StyleSheet, ViewProps, ViewStyle, StyleProp, TextStyle, I18nManager } from 'react-native';
 import { MD3Theme, Text, useTheme } from 'react-native-paper';
 import { $DeepPartial } from '@callstack/react-theme-provider';
-import { Icon } from '../Icon';
+import { Icon } from '../icon';
```
Comment (daileytj) [inline components/src/md3/ChannelValue/ChannelValue.tsx:5]: I just noticed the build is outputting the icon folder with a lowercase. Not sure why this is happening, but if this change works then we can roll with it.

File: components/src/md3/ChannelValue/ChannelValue.tsx [hunk 2]
```diff
@@ -110,7 +110,7 @@ export const ChannelValue: React.FC<ChannelValueProps> = (props) => {
     const getIcon = useCallback(() => {
         if (icon) {
             return (
-                <View style={[{ marginRight: Math.round(fontSize / 3) }]}>
+                <View style={[{ marginRight: Math.round(fontSize / 3), paddingTop: 2 }]}>
```
Comment (daileytj) [inline components/src/md3/ChannelValue/ChannelValue.tsx:113]: I don't think this is the appropriate fix for the icon alignment with the text. If you look at the showcase vs the MD3 demo you'll see that the showcase examples are now off. Not sure why we're getting different results, but showcase is where we should be testing things now and the icon is not aligning properly now.

Comment (daileytj) [reply components/src/md3/ChannelValue/ChannelValue.tsx:113]: The text seems to be aligned without any additional styles in the showcase

**Comment by daileytj** [comment]
I added a branch `feature/remove-unnecessary-md3-demo` to remove the demo and cleanup the icon file names and remove the ChannelValue padding. Feel free to look that over and possibly merge that into this branch if everything builds appropriately and you are able to work in the showcase. I added a showcase branch `feature/update-channelvalue-demo` as well to address the Channelvalue icon image example. 

We can address all these changes separately as well, but I think it would be good to get rid of that additional MD3 demo folder sooner than later.

**Comment by surajeaton** [comment]
> I added a branch `feature/remove-unnecessary-md3-demo` to remove the demo and cleanup the icon file names and remove the ChannelValue padding. Feel free to look that over and possibly merge that into this branch if everything builds appropriately and you are able to work in the showcase. I added a showcase branch `feature/update-channelvalue-demo` as well to address the Channelvalue icon image example.
> 
> We can address all these changes separately as well, but I think it would be good to get rid of that additional MD3 demo folder sooner than later.

Done. Please review.

**Comment by huayunh** [comment]
- Can you update the channel value to use a "semibold" font weight by default? Sorry I made some bad design decisions in the component sticker sheet.
- Are the CI test cases expected to fail? Or am I scope-creeping?
- To avoid legal troubles, ensure that the images you are using are copyright free. Talking about MD3/assets/images/test-avatar.png

**Comment by surajeaton** [comment]
> * Can you update the channel value to use a "semibold" font weight by default? Sorry I made some bad design decisions in the component sticker sheet.
> * Are the CI test cases expected to fail? Or am I scope-creeping?
> * To avoid legal troubles, ensure that the images you are using are copyright free. Talking about MD3/assets/images/test-avatar.png

- Updated the font weight in channel value.
- We might need to fix circle ci in separate pr, as it is failing due to various reasons, sometimes Install Chromedriver sometimes snapshots (even if the snapshots are updated using `yarn test -u` command)
- MD3/assets/images/test-avatar.png is taken from showcase of same repo

**Comment by huayunh** [comment]
> Updated the font weight in channel value

I saw that there's this code snippet 
```
                        {
                            color: getColor(),
                            fontSize: fontSize,
                            letterSpacing: 0,
                            fontFamily: 'OpenSans-Regular',
                            fontWeight: '600',
                        },
```
* Why is `letterSpacing` enforced here? 
* I see that we have both `fontFamily: 'OpenSans-Regular'` and `fontWeight: '600'` instead of just one `fontFamily: 'OpenSans-Semibold'` (or whatever the font is called). Is that a technical constraint?

> MD3/assets/images/test-avatar.png is taken from showcase of same repo

So are you sure that the showcase image was copyright-free to begin with? Perhaps @daileytj knows it better?

**Review by daileytj — DISMISSED** [review]

**Comment by daileytj** [comment]
@huayunh @surajeaton I updated the showcase and MD3 demo to use a public domain image

**Comment by surajeaton** [comment]
> * fontFamily: 'OpenSans-Semibold

Removed letterSpacing.
None of the fontFamily are updating the font weight (always 400) when passed as `fontFamily: 'OpenSans-Regular' or fontFamily: 'OpenSans-Semibold'.

We may need to consider it as an action item to fix it across the library.

**Comment by huayunh** [comment]
interesting...

I am not sure why MD3/components/ChannelValue.tsx was not getting updated, but the rest seems good to me

**Comment by daileytj** [comment]
> interesting...
> 
> I am not sure why MD3/components/ChannelValue.tsx was not getting updated, but the rest seems good to me

Everyone should be working from src/md3, I need to remove the MD3 folder entirely. I will do that after we get the rest of the currently worked-on components in.

**Comment by daileytj** [comment]
Simulator Screenshot - iPhone SE (3rd generation) - 2023-09-18 at 08 50 27
Simulator Screenshot - iPhone SE (3rd generation) - 2023-09-18 at 08 54 05
Simulator Screenshot - iPhone SE (3rd generation) - 2023-09-18 at 08 55 27
Simulator Screenshot - iPhone SE (3rd generation) - 2023-09-18 at 08 59 52

 For some reason the semi bold and the regular fonts look the same despite looking as though they are properly configured. At this point I say we leave it as is with regular font family and font weight 600 and address this at a later time.

**Review by daileytj — APPROVED** [review]

**Pull request merged**